### PR TITLE
datastore: admin samples

### PR DESCRIPTION
**Pull request by AlisskaPie**

### CODE REVIEW

File: datastore/admin/datastore_admin_client_create.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+func clientCreate(w io.Writer) error {
+	ctx := context.Background()
+	client, err := admin.NewDatastoreAdminClient(ctx)
+	if err != nil {
+		return fmt.Errorf("admin.NewDatastoreAdminClient: %v", err)
+	}
+	defer client.Close()
+
+	fmt.Fprintf(w, "Admin client created\n")
+	return nil
```
Comment (IlyaFaer) [inline datastore/admin/datastore_admin_client_create.go]: I think, we should return `client` here (and the things that matters in other samples)

File: datastore/admin/datastore_admin_entities_export.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	admin "cloud.google.com/go/datastore/admin/apiv1"
+	adminpb "google.golang.org/genproto/googleapis/datastore/admin/v1"
+)
+
+// entitiesExport exports a copy of all or a subset of entities from Google Cloud Datastore
+// to another storage system, such as Google Cloud Storage.
```
Comment (IlyaFaer) [inline datastore/admin/datastore_admin_entities_export.go]: This comments seem to be too long (>80 symbols)

File: datastore/admin/datastore_admin_entities_export.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+func entitiesExport(w io.Writer, projectID, outputURLPrefix string) error {
+	ctx := context.Background()
```
Comment (IlyaFaer) [inline datastore/admin/datastore_admin_entities_export.go:32]: `projectID` should be described here as a comment like:
```go
// projectID := "project-id"
```
And `outputURLprefix` as well

File: datastore/admin/datastore_admin_entities_export.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	defer client.Close()
+
+	req := &adminpb.ExportEntitiesRequest{
+		ProjectId:       projectID,
+		OutputUrlPrefix: outputURLPrefix}
+	op, err := client.ExportEntities(ctx, req)
+	if err != nil {
+		return fmt.Errorf("ExportEntities: %v", err)
+	}
+
```
Comment (IlyaFaer) [inline datastore/admin/datastore_admin_entities_export.go]: I'd erase this empty line and the next one

File: datastore/admin/datastore_admin_entities_import.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+	ctx := context.Background()
+	client, err := admin.NewDatastoreAdminClient(ctx)
+	if err != nil {
+		return fmt.Errorf("admin.NewDatastoreAdminClient: %v", err)
+	}
+	defer client.Close()
+
+	req := &adminpb.ImportEntitiesRequest{
+		ProjectId: projectID,
+		InputUrl:  inputURL}
```
Comment (IlyaFaer) [inline datastore/admin/datastore_admin_entities_import.go]: `}` in here is little bit strange. I assume it should be on the next line

**Review by IlyaFaer — COMMENTED** [review]
So, we don't have tests for import and export indexes. Is there any way to create indexes with Go code? I mean without Google Cloud Console.

File: datastore/admin/datastore_admin_entities_import.go (hunk 5 above)
Comment (AlisskaPie) [reply datastore/admin/datastore_admin_entities_import.go]: No, 'cause in that case there will be this error: 
`missing ',' before newline in composite literal`

File: datastore/admin/datastore_admin_entities_export.go (hunk 4 above)
Comment (AlisskaPie) [reply datastore/admin/datastore_admin_entities_export.go]: Done

File: datastore/admin/datastore_admin_entities_export.go (hunk 3 above)
Comment (AlisskaPie) [reply datastore/admin/datastore_admin_entities_export.go:32]: Done

File: datastore/admin/datastore_admin_entities_export.go (hunk 2 above)
Comment (AlisskaPie) [reply datastore/admin/datastore_admin_entities_export.go]: Done

File: datastore/admin/datastore_admin_client_create.go (hunk 1 above)
Comment (AlisskaPie) [reply datastore/admin/datastore_admin_client_create.go]: ok, done

**Comment by AlisskaPie** [comment]
> So, we don't have tests for import and export indexes. Is there any way to create indexes with Go code? I mean without Google Cloud Console.

Unfortunately no, I've found in [this document ](https://cloud.google.com/datastore/docs/reference/admin/rpc/google.datastore.admin.v1#createindexrequest)method `CreateIndex`, but it seems like it's not implemented yet.

File: datastore/admin/datastore_admin_client_create.go [hunk 6]
```diff
@@ -24,16 +24,16 @@ import (
...
 // clientCreate creates a new Datastore admin client.
-func clientCreate(w io.Writer) error {
+func clientCreate(w io.Writer) (*admin.DatastoreAdminClient, error) {
 	ctx := context.Background()
 	client, err := admin.NewDatastoreAdminClient(ctx)
 	if err != nil {
-		return fmt.Errorf("admin.NewDatastoreAdminClient: %v", err)
+		return nil, fmt.Errorf("admin.NewDatastoreAdminClient: %v", err)
 	}
 	defer client.Close()
```
Comment (IlyaFaer) [inline datastore/admin/datastore_admin_client_create.go]: `defer` in create function, like yesterday's one we figured out

File: datastore/admin/datastore_admin_entities_export.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+// Cloud Datastore to another storage system, such as Google Cloud Storage.
+func entitiesExport(w io.Writer, projectID, outputURLPrefix string) error {
+	// projectID := "project-id"
+	// outputURLPrefix := "gs://bucket-name"
+	ctx := context.Background()
+	client, err := admin.NewDatastoreAdminClient(ctx)
+	if err != nil {
+		return fmt.Errorf("admin.NewDatastoreAdminClient: %v", err)
+	}
+	defer client.Close()
```
Comment (IlyaFaer) [inline datastore/admin/datastore_admin_entities_export.go:37]: On this line `defer` is okay

File: datastore/admin/datastore_admin_entities_import.go (hunk 5 above)
Comment (IlyaFaer) [reply datastore/admin/datastore_admin_entities_import.go]: You can add `,` after `inputURL`

**Review by IlyaFaer — COMMENTED** [review]
> Unfortunately no, I've found in this document method CreateIndex, but it seems like it's not implemented yet.

Hm-m, we should ask about it googlers with public PR

File: datastore/admin/datastore_admin_entities_import.go (hunk 5 above)
Comment (AlisskaPie) [reply datastore/admin/datastore_admin_entities_import.go:40]: This wouldn't work

**Review by IlyaFaer — APPROVED** [review]

**Pull request closed without merging**